fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/wallet/ContentWalletFragment.java

```diff
@@ -51,38 +51,37 @@ public View onCreateView(
     mWalletContentBinding.setViewmodel(mHomeViewModel);
     mWalletContentBinding.setLifecycleOwner(requireActivity());
 
-    return mWalletContentBinding.getRoot();
-  }
-
-  @Override
-  public void onActivityCreated(@Nullable Bundle savedInstanceState) {
-    super.onActivityCreated(savedInstanceState);
-
     setupListAdapter();
     setupListUpdates();
 
     if (wallet.isSetUp()) {
-      mWalletContentBinding.logoutButton.setVisibility(View.VISIBLE);
-      mWalletContentBinding.logoutButton.setOnClickListener(
-          clicked -> {
-            if (logoutAlert != null && logoutAlert.isShowing()) {
-              logoutAlert.dismiss();
-            }
-            AlertDialog.Builder builder = new AlertDialog.Builder(requireActivity());
-            builder.setTitle(R.string.logout_title);
-            builder.setMessage(R.string.logout_message);
-            builder.setPositiveButton(
-                R.string.confirm,
-                (dialog, which) -> {
-                  mHomeViewModel.logoutWallet();
-                  WalletFragment.openWallet(
-                      getParentFragmentManager(), mHomeViewModel.isWalletSetUp());
-                });
-            builder.setNegativeButton(R.string.cancel, (dialog, which) -> dialog.dismiss());
-            logoutAlert = builder.create();
-            logoutAlert.show();
-          });
+      setupWalletContent();
     }
+
+    return mWalletContentBinding.getRoot();
+  }
+
+  private void setupWalletContent() {
+    mWalletContentBinding.logoutButton.setVisibility(View.VISIBLE);
+    mWalletContentBinding.logoutButton.setOnClickListener(
+        clicked -> {
+          if (logoutAlert != null && logoutAlert.isShowing()) {
+            logoutAlert.dismiss();
+          }
```

Comment:
why dismiss it if it's already shown, and then re-create it ? 
Why not just `return` ?

Reply:
I... don't know

Let's just return there